src/combinatorics/simple_graph/basic.lean

```diff
@@ -347,21 +347,25 @@ are adjacent in the complement, and every nonadjacent pair of vertices is adjace
 -/
 def complement (G : simple_graph V) : simple_graph V :=
 { adj := λ v w, v ≠ w ∧ ¬G.adj v w,
-  sym := λ v w h,
-    begin
-      cases h with hne hnadj,
-      split,
-      symmetry,
-      exact hne,
-      by_contra,
-      apply hnadj (G.sym h),
-    end,
-  loopless := λ v,
-    begin
-      by_contra,
-      cases h with hne hnadj,
-      apply hne rfl,
-    end }
+  sym := λ v w ⟨hne, _⟩, ⟨hne.symm, by rwa edge_symm⟩,
+  loopless := λ v ⟨hne, _⟩, false.elim (hne rfl) }
+
+/--
+The complement of the complement of `G` is equal to `G`.
+-/
+lemma complement_invol (G : simple_graph V) : complement (complement G) = G :=
+begin
+  ext v w,
+  split,
+  { intros h,
+    simp_rw [complement, not_and, not_not] at h,
+    exact h.2 h.1 },
+  { intros h,
+    simp only [complement, not_and, not_not],
+    refine ⟨_, λ h2, h⟩,
+    { rintro rfl,
```

Comment:
Isn't there `ne_of_adj` for this sort of thing, maybe?

Reply:
Sorry @agusakov for all these code golfs, it's just kind of satisfying tersifying proofs.
```lean
begin
  ext v w,
  split; simp only [compl_adj, not_and, not_not],
  { exact λ ⟨hne, h⟩, h hne },
  { intro h,
    simpa [G.ne_of_adj h], },
end
```

Reply:
I really need to get better at golfing, I appreciate it

Reply:
Also yeah there's `ne_of_adj`, forgot about that. Ty @b-mehta